text/0000-app-mixins.md

```diff
@@ -0,0 +1,305 @@
+# Meta
+[meta]: #meta
+- Name: Application Mixins
+- Start Date: 2020-06-22
+- Author(s): [Joe Kutner](https://github.com/jkutner/)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for allowing application developers (buildpack users) to specify mixins that will be dynamically included with their build and run images.
+
+# Motivation
+[motivation]: #motivation
+
+Mixins already allow buildpack authors to create buildpacks that depend on an extended set of OS packages without affecting build time, but it's common for application code to depend on OS packages too. But allowing buildpacks to arbitrarily install OS packages during build would drastically increase build time, especially when CNB tooling is used to build or rebase many apps with similar package requirements.
+
+For that reason, this proposal defines a mechanism to implement dynamic installation of mixins at build time while minimizing the performance impact. This is accomplished by allowing stack images to satisfy mixin requires statically and/or reject dynamic mixin installation early in the build process.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- *application developer* - a person who is using buildpacks to transform their application code into an OCI image
+- *root buildpack* - a new type of buildpack that runs as the root user
+- *stackpack* - a type of root buildpack that runs against the stack image(s) instead of an app. It is distinguished by a static list of mixins it can provide.
```

Comment:
```suggestion
- *stack buildpack* - a type of root buildpack that runs against the stack image(s) instead of an app. It is distinguished by a static list of mixins it can provide.
```

I propose keeping the naming more inline with "buildpacks" and simply prefix the type as the name as a whole. Specially, if we can envision being able to provide them at build time from a platform's perspective.